src/command_modules/azure-cli-monitor/azure/cli/command_modules/monitor/custom.py

```diff
@@ -97,3 +98,90 @@ def _validate_start_time(start_time, end_time):
         raise ValueError("start_time '{}' is later than Now {}.".format(start_time, now))
 
     return result_time
+
+
+def list_activity_logs(client, correlation_id=None, resource_group=None, resource_uri=None,
+                       resource_provider=None, start_time=None, end_time=None,
+                       caller=None, status=None, max_events=50, select=None):
+    '''Provides the list of activity logs.
+    :param str correlation_id: The correlation id of the query
+    :param str resource_group: The resource group
+    :param str resource_uri: The identifier of the resource
+    :param str resource_provider: The resource provider
+    :param str start_time: The start time of the query. In ISO format with explicit indication of
+                           timezone: 1970-01-01T00:00:00Z, 1970-01-01T00:00:00-0500
+    :param str end_time: The end time of the query. In ISO format with explicit indication of
+                         timezone: 1970-01-01T00:00:00Z, 1970-01-01T00:00:00-0500
+    :param str caller: The caller to look for when querying
+    :param str status: The status value to query (ex: Failed)
+    :param str max_events: The maximum number of records to be returned by the command
+    :param str select: The list of event names
+    '''
+    present_switch = 0
+    for param in [correlation_id, resource_group, resource_uri, resource_provider]:
+        present_switch = present_switch + (1 if param else 0)
+
+    if present_switch > 1:
+        raise CLIError("'--correlation-id', '--resource-group', '--resource-uri'"
+                       " and '--resource-provider' are optional and mutually exclusive.")
+
+    formatter = "eventTimestamp ge {} and eventTimestamp le {}"
+    odata_filters = _validate_time_range_and_add_defaults(start_time, end_time,
+                                                          formatter=formatter)
+
+    if correlation_id:
+        odata_filters = _process_prameter(odata_filters, 'correlation_id',
```

Comment:
Typo: _process_parameter

Additionally that is not a very descriptive name. Perhaps something like "_build_odata_filter"?